### DISCUSSION TITLE
Mismatch of main heatmap row names and annotation values

### DISCUSSION THREAD

**Discussion by planetypus** [jokergoo/ComplexHeatmap#1035] [General]
Hello. I recently discovered this wonderful tool. Thank you.
I am trying to annotate a heatmap with a barplot at the right side. 
My matrix has 253 variables and 202 observations.
I created three barplot annotations at the top, and it was working fine.
I also tried to make another bargraph at the left side. At first glance, the visualization seemed fine, but when I checked closer, the values did not match the row names. The values were interchanged all over the place, like a messed up indexing between row names and values.
I did a kmeans clustering for both rows and columns. 
I wonder if I need to perform indexing of the row names to match with my annotation value. If so, may I know how to index my value to match with the row names?
Or, is there any better way to do it?

I would REALLY appreciate any help! Thank you very much!

**Comment by jokergoo** [comment]
The row indices should always be the same for the matrix and for your barplots. The reordering is done by the package automatically.
